compass/ocean/tests/global_convergence/cosine_bell/__init__.py

```diff
@@ -0,0 +1,104 @@
+            The global ocean test group that this test case belongs to
+        """
+        super().__init__(test_group=test_group, name='cosine_bell')
+        self.resolutions = None
+
+    def configure(self):
+        """
+        Set config options for the test case
+        """
+        config = self.config
+        resolutions = config.get('cosine_bell', 'resolutions')
+        resolutions = [int(resolution) for resolution in
+                       resolutions.replace(',', ' ').split()]
+
+        self.resolutions = resolutions
+
+        init_options = dict()
+        for option in ['temperature', 'salinity', 'lat_center', 'lon_center',
+                       'radius', 'psi0', 'vel_pd']:
+            init_options['config_cosine_bell_{}'.format(option)] = \
+                config.get('cosine_bell', option)
+
+        for resolution in resolutions:
+            self.add_step(Mesh(test_case=self, resolution=resolution))
+
+            step = Init(test_case=self, resolution=resolution)
+            step.add_namelist_options(options=init_options, mode='init')
+            self.add_step(step)
+
+            self.add_step(Forward(test_case=self, resolution=resolution))
+
+        self.add_step(Analysis(test_case=self, resolutions=resolutions))
+
+        self.update_cores()
+
+    def run(self):
+        """
+        Run each step of the testcase
+        """
+        config = self.config
+        for resolution in self.resolutions:
+            cores = config.getint('cosine_bell',
+                                  'QU{}_cores'.format(resolution))
+            min_cores = config.getint('cosine_bell',
+                                      'QU{}_min_cores'.format(resolution))
+            step = self.steps['QU{}_forward'.format(resolution)]
+            step.cores = cores
+            step.min_cores = min_cores
+
+        # run the step
+        super().run()
+
+    def update_cores(self):
+        """ Update the number of cores and min_cores for each forward step """
+
+        config = self.config
+
+        goal_cells_per_core = config.getfloat('cosine_bell',
+                                              'goal_cells_per_core')
+        max_cells_per_core = config.getfloat('cosine_bell',
+                                             'max_cells_per_core')
+
+        for resolution in self.resolutions:
+            # a heuristic based on QU30 (65275 cells) and QU240 (10383 cells)
+            approx_cells = 6e8 / resolution**2
+            # ideally, about 300 cells per core
+            # (make it a multiple of 4 because...it looks better?)
+            cores = max(1,
+                        4*round(approx_cells / (4 * goal_cells_per_core)))
+            # In a pinch, about 3000 cells per core
+            min_cores = max(1,
+                            round(approx_cells / max_cells_per_core))
+            step = self.steps['QU{}_forward'.format(resolution)]
+            step.cores = cores
+            step.min_cores = min_cores
```

Comment:
@vanroekel, could you let me know if you're okay with this heuristic for computing core counts fore each resolution? The result is:
```
qu60_cores = 556
qu60_min_cores = 56
qu90_cores = 248
qu90_min_cores = 25
qu120_cores = 140
qu120_min_cores = 14
qu150_cores = 88
qu150_min_cores = 9
qu180_cores = 60
qu180_min_cores = 6
qu210_cores = 44
qu210_min_cores = 5
qu240_cores = 36
qu240_min_cores = 3
```

Reply:
yes this looks like a good way to approximate cores to make this test more efficient